### PR TITLE
Use relative url instead of static urls

### PR DESCRIPTION
**Pull request by popojargo**
## Overview

* Add helper functions to get API url and Server urls
* Fix some url who were not using relative urls
* Fix documentation urls that were using _utils instead of relative
* Update test according to change
* Changed default configuration in settings.json

## Testing recommendations

I used a express pouchdb setup.

1. Create a folder 👍 
2. In that folder create a package.json

```json
{
  "name": "test",
  "version": "1.0.0",
  "description": "",
  "main": "test.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "author": "",
  "license": "ISC",
  "dependencies": {
    "express": "^4.16.3",
    "express-pouchdb": "^4.0.0",
    "morgan": "^1.4.0",
    "pouchdb": "^6.4.3"
  }
}
```

3. Create a test.js

```javascript
var express = require('express')
    , app = express()
    , PouchDB = require('pouchdb')
    , logger = require('morgan')
;

app.use(logger("tiny"));
app.get('/', function (req, res, next) {
    res.send('Welcome Traveler!  <a href="/db/_utils/">Fauxton?</a>');
});
//app.use('/express-pouchdb', require('express-pouchdb')(PouchDB));
app.use('/db', require('express-pouchdb')(PouchDB));
app.listen(3000);
```

4. Install depencies with `npm install`

5. You need to edit `node_modules/express-pouchdb/lib/routes/fauxton.js` and change the exports for:

```javascript
module.exports = function (app) {
  app.get('/_utils/', function (req, res) {
    // We want to force /_utils to /_utils/ as this is the CouchDB behavior
    // https://git.io/vDMOD#L78
    // https://git.io/vDMOQ#L974
    if (req.originalUrl === '/_utils') {
      res.redirect(301, '/_utils/');
    } else {
      res.sendFile(path.normalize(FAUXTON_PATH + '/index.html'));
    }
  });

  app.use('/_utils/', express.static(FAUXTON_PATH));
};
```

6. Make a fauxton release (`grunt release`) and copy the release to `node_modules/pouchdb-fauxton/www`

7. You need to replace in the index.html of the release (`dashboard` -> `./dashboard`)

8. You should be good 💃 

## GitHub issue number
#944 

## Checklist

- [ ] Code is written and works correctly;
- [ ] Changes are covered by tests;
- [ ] Documentation reflects the changes;
- [ ] Update [rebar.config.script](https://github.com/apache/couchdb/blob/master/rebar.config.script) with the correct tag once a new Fauxton release is made

### CODE REVIEW

File: app/constants.js [hunk 1]
```diff
@@ -29,23 +28,23 @@ export default {
...
-    DB_PERMISSION: '/_utils/docs/api/database/security.html#db-security',
-    STATS: '/_utils/docs/api/server/common.html?highlight=stats#get--_stats',
-    ACTIVE_TASKS: '/_utils/docs/api/server/common.html?highlight=stats#active-tasks',
-    LOG: '/_utils/docs/api/server/common.html?highlight=stats#log',
-    CONFIG: '/_utils/docs/config/index.html',
-    VIEWS: '/_utils/docs/intro/overview.html#views',
-    MANGO_INDEX: '/_utils/docs/intro/api.html#documents',
-    MANGO_SEARCH: '/_utils/docs/intro/api.html#documents',
-    CHANGES: '/_utils/docs/api/database/changes.html?highlight=changes#post--db-_changes'
+    GENERAL:'./docs/intro/api.html#documents',
```
Comment (popojargo) [inline app/constants.js]: I should probably use the getServerUrl here to be more consistent but I had trouble with circular dependencies 😞

File: app/helpers.js [hunk 2]
```diff
@@ -63,6 +64,22 @@ Helpers.escapeJQuerySelector = function (selector) {
   return selector && selector.replace(/[!"#$%&'()*+,./:;<=>?@[\\\]^`{|}~]/g, "\\$&");
 };
 
+Helpers.getApiUrl = endpointRoute => {
+  return url.resolve(window.location.href, app.host + endpointRoute);
+};
+
+Helpers.getServerUrl = endpointRoute => {
+  if (endpointRoute.startsWith("/")) {
```
Comment (popojargo) [inline app/helpers.js]: I remove the first slash to avoid having `//_session`

File: settings.json.default.json [hunk 3]
```diff
@@ -27,7 +27,7 @@
         },
         "app": {
           "root": "/",
-          "host": "../..",
+          "host": "./",
```
Comment (popojargo) [inline settings.json.default.json]: In dev, we have a proxy at the root

File: settings.json.default.json [hunk 4]
```diff
@@ -42,7 +42,7 @@
         },
         "app": {
           "root": "/",
-          "host": "../..",
+          "host": "../",
```
Comment (popojargo) [inline settings.json.default.json]: In a release, Fauxton should be at `/_utils/` so we have to get back one "folder".

**Comment by wohali** [comment]
+1 on this if you can rebase it

**Comment by popojargo** [comment]
I will compile Couch with this branch and see if there's not breaking change. Then I'll merge when Travis goes green

**Pull request merged**